src/fragments/lib/logging/common/remote_configuration/remote_configuration.mdx

```diff
@@ -0,0 +1,220 @@
+
+    const apiInvokePolicy = new iam.PolicyStatement({
+      effect: iam.Effect.ALLOW,
+      resources: [ getLoggingConstraints.methodArn ],
+      actions: ['execute-api:Invoke']
+    })
+
+    authRole.addToPrincipalPolicy(apiInvokePolicy)
+    unAuthRole.addToPrincipalPolicy(apiInvokePolicy)
+
+  }
+}
+```
+
+## Sample Lambda handler
+Below is a sample lambda that reads and returns the `remoteloggingconstraints.json` from AWS S3.
+Note that caching and ETag is used.  To reduce data usage, if the logging levels has not changed since the last fetch, there is no need to return the log levels.
+
+```js
+const { S3Client, GetObjectCommand } = require("@aws-sdk/client-s3")
+const s3 = new S3Client({})
+const bucketName = process.env.BUCKET
+const key = process.env.KEY
+let cachedConfig = {
+    expiresOn: 0, 
+    ETag: '', 
+    config: ''
+}
+
+exports.main = async function(event, context) {
+  try {
+    if (event.httpMethod === "GET") {
+      if (!cachedConfig.config || Date.now() > cachedConfig.expiresOn) {
+        // refresh cache if cache is invalid
+        const command = new GetObjectCommand({ Bucket: bucketName, Key: key })
+        const s3Resp = await s3.send(command)
+        await setCachedConfig(s3Resp)
+      }
+
+      if (event.headers['If-None-Match'] === cachedConfig.ETag) {
+        // return 304 not modified if config has not changed
+        return {
+          statusCode: 304,
+        }
+      } else {
+        // return updated/modified config with latest ETag
+        return {
+          statusCode: 200,
+          headers: {"'If-None-Match'": cachedConfig.ETag },
+          body: cachedConfig.config
+        }
+      }
+    }
+  } catch(error) {
+    const resp = error.stack || JSON.stringify(error, null, 2)
+    return {
+      statusCode: 400,
+        headers: {},
+        body: JSON.stringify(resp)
+    }
+  }
+}
+
+const setCachedConfig = async (s3Resp) => {
+  cachedConfig = {
+    expiresOn: Date.now() + 600 * 1000, //10 minutes
+    ETag: s3Resp.ETag.replace(/\"/gi, ''), //remove \" from string
+    config: await s3Resp.Body.transformToString()
+  }
+}
+```
+
+## Sample `remoteloggingconstraints.json`
+Below is a sample remote config file that overwrites the local file in the mobile application.  This file will be deployed to S3.  Once deployed, you can change your application log levels by editing this file in S3.
+
+```json
+{
+    "defaultLogLevel": "ERROR",
+    "categoryLogLevel": {
+        "API": "DEBUG",
+        "AUTH": "DEBUG",
+    },
+    "userLogLevel": {
+        "cognito-sub-xyz-123": {
+            "defaultLogLevel": "VERBOSE",
+            "categoryLogLevel": {
+                "API": "VERBOSE",
+                "AUTH": "VERBOSE",
+            }
+        }
+    }
+}
+```
+
+## Enable Remote Log Levels
+To enable Amplify Logger to fetch remote log levels, you will need to provide the api endpoint that has the log levels and the fresh interval that you want the Amplify Logger to refresh.
```

Comment:
the API endpoint.

fresh interval? what does that mean? Maybe refresh?

Reply:
fixed typo and spelling mistake.